app/client/views/home/activities/sums/sums.js

```diff
@@ -1,59 +1,86 @@
-Template.homeResidentActivitySumsByType.created = function () {
+Template.homeResidentActivitySumsByType.onCreated(function () {
   // Get reference to template instance
-  var instance = this;
+  const templateInstance = this;
 
   // Get current home ID
-  instance.homeId = Router.current().params.homeId;
+  templateInstance.homeId = Router.current().params.homeId;
 
   // set up home resident activity sums by type reactive variable
- instance.homeResidentsActivitySumsByType = new ReactiveVar();
+ templateInstance.homeResidentsActivitySumsByType = new ReactiveVar();
 
   // Get home resident activity sums from server method
-  Meteor.call('getHomeResidentsActivitySumsByTypeLast30Days', instance.homeId, function (error, activitySums) {
+  Meteor.call('getHomeResidentsActivitySumsByTypeLast30Days', templateInstance.homeId, function (error, activitySums) {
     //Set the home resident activity sums variable with the returned activity sums
-    instance.homeResidentsActivitySumsByType.set(activitySums);
+    templateInstance.homeResidentsActivitySumsByType.set(activitySums);
   });
-};
+});
 
-Template.homeResidentActivitySumsByType.rendered = function () {
+Template.homeResidentActivitySumsByType.onRendered(function () {
   // Get reference to template instance
-  var instance = this;
+  const templateInstance = this;
 
-  var activityCountChart = nv.models.multiBarHorizontalChart();
-
-  // Get colors from D3, twenty category
-  var colors = d3.scale.category20().range();
-
-  // Remove red from the colors list
-  var customColors = _.without(colors, "#d62728");
-
-  activityCountChart
-    //.height(800)
-    .x(function(d) { return d.label })
-    .y(function(d) { return d.value })
-    .duration(250)
-    .margin({bottom: 25, left: 70})
-    .stacked(true)
-    .showControls(false)
-    .color(customColors)
-
-  activityCountChart
-    .yAxis.axisLabel("Activity count")
-    .tickFormat(d3.format('d'));
-
-  instance.autorun(function () {
+  templateInstance.autorun(function () {
```

Comment:
Make sure the chart rendering autorun is within the `Template.onRendered` callback, for safety.

Reply:
What do you mean? 

1) It's the general part in whole project where chart is rendered
2) Before rendering it makes sure data exists

Please help me What do I need to check additionally?

Reply:
IIs this in an onCreated or onRendered autorun definition?

Reply:
If the chart-render autorun is defined in an onCreated callback, just move that autorun to an onRendered callback. That will ensure DOM is ready, even though the data method will be slow. Just setting up a good habit: anything that relies on DOM should be guaranteed to run after DOM is ready.

Reply:
Sorry, I was reading this on mobile and didn't see the whole file.